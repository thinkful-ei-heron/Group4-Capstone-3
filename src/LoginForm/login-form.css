.LoginForm .alert{
	height: 3em;
	max-height: 3em;
	min-height: 3em;
}
.login-page {
	background-image: url("../images/beer3.jpg");
	background-repeat: no-repeat;
	background-position: center;
	background-size: cover;
	min-height: 100vh;
	display: flex;
	justify-content: center;
}
.login {
	background-color: rgba(0, 0, 0, 0.584);
	padding: 40px 20px;
	align-self: center;
}
.login-page h1 {
	font-family: 'Merriweather', serif;
	color: white;
	font-size: 3em;
}
.login-page h3 {
	font-family: 'Roboto Slab', serif;
	color: white;
	font-size: 2em;
}
.login-header {
	text-decoration: none;
	cursor: pointer;
}



.login-page input {
	background-color: rgba(255, 255, 255, 0.66);
	font-size: 20px;
	color: black;
	margin-bottom: 20px;
}

.login-page button {
	background-color: transparent;
	border-radius: 0px;
	color: white;
	font-size: 20px;
	cursor: hover;
}

.submit-btn {
	background: rgb(112, 206, 99);
	color: white;
	border-style: outset;
	border-color: #0066a2;
	height: 40px;
	font: bold 15px arial, sans-serif;
	text-shadow: none;
	cursor: pointer;
	width: 6em;
	font-family: 'Merriweather', serif;
}
.submit-btn:hover {
	background-color: rgba(0, 0, 0, 0.384);
	color: white;
	border-style: outset;
	border-color: #0066a2;
}

.other-btns {
	margin-top: 50px;
	display: flex;
	justify-content: space-evenly;
	cursor: hover;
}

.new-user-btn {
	border: none;
	cursor: pointer;
	font-family: 'Merriweather', serif;
}
.new-user-btn:hover {
	color: rgb(252, 189, 55);
}
/* @media only screen and (max-width: 1024px) {
} */
@media (min-width: 480px) {
	.login {
		padding: 60px 30px;
	}
	.login-page h1 {
		font-size: 4em;
	}
	.login-page h3 {
		font-size: 3em;
	}
}
@media (min-width: 767px) {
	.login {
		padding: 90px 50px;
	}
	.login-page h1 {
		font-size: 5em;
	}
	.login-page h3 {
		font-size: 4em;
	}
}
@media (min-width: 480px) and (orientation: landscape) {
	.login {
		padding: 30px 20px;
	}
}
@media (min-width: 767px) and (orientation: landscape) {
	.login {
		padding: 40px 30px;
	}
}
