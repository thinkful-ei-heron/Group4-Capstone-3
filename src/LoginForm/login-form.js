import React, { Component } from 'react';
import { Link } from 'react-router-dom'
import UserContext from "../contexts/UserContext";
import AuthApiService from "../services/auth-api-service";
import './login-form.css' 

export default class LoginForm extends Component {
  handleLoginSuccess = () => {
    const { location, history } = this.props;
    const destination = (location.state || {}).from || "/home";
    history.push(destination);
  };

  static contextType = UserContext;

  state = { error: null };

  firstInput = React.createRef();

  handleSubmit = ev => {
    ev.preventDefault();
    const { username, password } = ev.target;

    this.setState({ error: null });

    AuthApiService.postLogin({
      user_name: username.value,
      password: password.value,
    })
      .then(res => {
        username.value = "";
        password.value = "";
        this.context.processLogin(res.authToken);
        this.handleLoginSuccess();
      })
      .catch(res => {
        this.setState({ error: res.error });
      });
  };

  render() {
    const { error } = this.state;
    return (
      <div className='login-page'>
        <div className='login'>
        <h1>Dear Beer</h1>
        <h3>Login</h3>
        <form onSubmit={this.handleSubmit}>
          <div role='alert'>
            {error && <p className='error'>Something went wrong!</p>}
          </div>
            <section>
              <div>
                <label htmlFor='login-username'>
                  {/* Username */}
                </label>
                <input 
                  type='text'
                  id='login-username'
                  name='username'
                  required
                  placeholder='Username'/>
              </div>
              <div>
                <label htmlFor='login-password'>
                  {/* Password */}
                </label>
                <input 
                  type='password'
                  id='login-password'
                  name='password'
                  required
                  placeholder='Password'/>
              </div>
              <div>
                <button
                  type='submit'>
                    Login
                </button>
                <Link to='/'>
                    <button>
                        Back
                    </button>
                </Link>
                <Link
                  to='/register'>
                    <button>
                        New user?
                    </button>
                </Link>
              </div>
            </section>
        </form>
      </div>
      </div>
    )
  }
}