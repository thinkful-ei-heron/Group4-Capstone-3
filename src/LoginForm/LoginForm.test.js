import React from 'react';
import ReactDOM from 'react-dom';
import LoginForm from './login-form'
import { shallow } from 'enzyme'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'
import { BrowserRouter } from 'react-router-dom';

Enzyme.configure({ adapter: new Adapter() });

describe('Login Form Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render
      (<BrowserRouter> 
          <LoginForm /> 
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('Displays LoginForm when rendered', () => {
    const wrapper = shallow(<LoginForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
});