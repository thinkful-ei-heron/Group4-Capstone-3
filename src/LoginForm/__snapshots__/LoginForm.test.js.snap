// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form Component Displays LoginForm when rendered 1`] = `
<div
  className="login-page"
>
  <div
    className="login"
  >
    <Link
      className="login-header"
      to="/"
    >
      <h1>
        DearBeer
      </h1>
    </Link>
    <h3>
      Login
    </h3>
    <form
      className="LoginForm"
      onSubmit={[Function]}
    >
      <div
        className="alert"
        role="alert"
      />
      <section>
        <div>
          <label
            htmlFor="login-username"
          />
          <input
            id="login-username"
            name="username"
            placeholder="Username"
            required={true}
            type="text"
          />
        </div>
        <div>
          <label
            htmlFor="login-password"
          />
          <input
            id="login-password"
            name="password"
            placeholder="Password"
            required={true}
            type="password"
          />
        </div>
        <div>
          <button
            className="submit-btn"
            type="submit"
          >
            Submit
          </button>
          <div
            className="other-btns"
          >
            <Link
              to="/register"
            >
              <button
                className="new-user-btn"
              >
                New user?
              </button>
            </Link>
          </div>
        </div>
      </section>
    </form>
  </div>
</div>
`;
